src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -1,771 +1,53 @@
-                }
-            }
-          }
-        } else Future.successful(ids)
-
-      logger.info("Ready to fetch updated delegates information from the chain")
-
-      /* Streams the (unevaluated) incoming data, actually fetching the results.
-       * We use combinators to keep the ongoing requests' flow under control, taking advantage of
-       * akka-streams automatic backpressure control.
-       * The results are grouped to optimize for database storage.
-       * We do this to re-aggregate results from pages which are now based on single blocks,
-       * which would lead to inefficient storage performances as-is.
-       */
-      val saveDelegates = (pages: LazyPages[tezosNodeOperator.DelegateFetchingResults]) =>
-        Source
-          .fromIterator(() => pages)
-          .mapAsync(1)(identity) //extracts the future value as an element of the stream
-          .mapConcat(identity) //concatenates the list of values as single-valued elements in the stream
-          .grouped(batchingConf.blockPageSize) //re-arranges the process batching
-          .mapAsync(1)(processDelegatesPage)
-          .runFold(Monoid[Int].empty) { (processed, justDone) =>
-            processed |+| justDone
-          } andThen logOutcome
-
-      val fetchAndStore = for {
-        (delegatesPages, _) <- prunedUpdates().map(tezosNodeOperator.getDelegatesForBlocks)
-        _ <- saveDelegates(delegatesPages) flatTap cleanup
-      } yield Done
-
-      fetchAndStore.transform(
-        identity,
-        e => {
-          val error = "I failed to fetch delegates from client and update them"
-          logger.error(error, e)
-          DelegatesProcessingFailed(message = error, e)
-        }
-      )
-
-    }
-  }
-
-  /** Keeps track of time passed between different partial checkpoints of some entity processing
-    * Designed to be partially applied to set properties of the whole process once, and then only compute partial completion
-    *
-    * @param entityName a string that will be logged to identify what kind of resource is being processed
-    * @param totalToProcess how many entities there were in the first place
-    * @param processStartNanos a nano-time from jvm monotonic time, used to identify when the whole processing operation began
-    * @param processed how many entities were processed at the current checkpoint
-    */
-  private[this] def logProcessingProgress(entityName: String, totalToProcess: Int, processStartNanos: Long)(
-      processed: Int
-  ): Unit = {
-    val elapsed = System.nanoTime() - processStartNanos
-    val progress = processed.toDouble / totalToProcess
-    logger.info("================================== Progress Report ==================================")
-    logger.info("Completed processing {}% of total requested {}s", "%.2f".format(progress * 100), entityName)
+    val program = Parallel.parProduct(indexingLoop, bakerRightsSchedule).void
```

Comment:
This simply combines the Kleisli functions as two process descriptions that will run in parallel when the `program` combined IO description will be executed by the `IOApp` object